src/services/node/NodeTransactionPoolService.ts

```diff
@@ -14,24 +14,103 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import { CalcFee } from '@substrate/calc';
+import BN from 'bn.js';
+import { BadRequest } from 'http-errors';
+
 import { INodeTransactionPool } from '../../types/responses';
 import { AbstractService } from '../AbstractService';
 
 export class NodeTransactionPoolService extends AbstractService {
-	async fetchTransactionPool(): Promise<INodeTransactionPool> {
+	async fetchTransactionPool(
+		includeFee: boolean
+	): Promise<INodeTransactionPool> {
 		const { api } = this;
 
 		const extrinsics = await api.rpc.author.pendingExtrinsics();
 
-		const pool = extrinsics.map((ext) => {
+		if (includeFee) {
+			const calcFeeWithParams = await this.getFromParams();
+			const weight = this.getWeight();
+
+			if (!calcFeeWithParams) {
+				throw new BadRequest(
+					'This runtime does not have the sufficient materials to calculate fees.'
+				);
+			}
+
+			return {
+				pool: extrinsics.map((ext) => {
+					const { hash, encodedLength, tip } = ext;
+					const partialFee = calcFeeWithParams.calc_fee(
+						// TODO This value comes from the success, failure events.
+						BigInt(0),
```

Comment:
This is the last piece of this calculation to figure out. Originally this value comes from the weight provided by a failed or successful event.

Reply:
Might make more sense to use `api.rpc.payment.queryInfo` to find fee estimate

Reply:
Okay so this is the right move, I am going to remove all `@substrate/calc` logic here and replace it with `queryInfo`.